File: libs/backend/domain/src/main/java/reengineering/ddd/teamai/model/Context.java
package reengineering.ddd.teamai.model;

import reengineering.ddd.archtype.Entity;
import reengineering.ddd.teamai.description.ContextDescription;

public class Context implements Entity<String, ContextDescription> {
  private String identity;
  private ContextDescription description;

  public Context(String identity, ContextDescription description) {
    this.identity = identity;
    this.description = description;
  }

  private Context() {
  }

  @Override
  public String getIdentity() {
    return this.identity;
  }

  @Override
  public ContextDescription getDescription() {
    return this.description;
  }
}

File: libs/frontend/domain/src/lib/model/context.ts
import { Entity } from '../archtype/entity.js';
import { ContextDescription } from '../description/index.js';

export class Context implements Entity<string, ContextDescription> {
  constructor(
    private identity: string,
    private description: ContextDescription
  ) {}

  getIdentity(): string {
    return this.identity;
  }
  getDescription(): ContextDescription {
    return this.description;
  }
}
